src/mongocxx/client.cpp

```diff
@@ -55,6 +55,16 @@ void* client::implementation() const {
     return _impl->client_t;
 }
 
+void client::read_concern(class read_concern rc) {
+    libmongoc::client_set_read_concern(_impl->client_t, rc._impl->read_concern_t);
+}
+
+class read_concern client::read_concern() const {
+    auto rc = libmongoc::client_get_read_concern(_impl->client_t);
+    return (class read_concern){
```

Comment:
I think the copy's fine.  The point of these things is that you set the read/write concern at some level and get re-use that way, not that the objects are magically cheap